packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -283,3 +287,27 @@ class NoopTreeLoader implements ITreeLoader {
     return EMPTY;
   }
 }
+
+class TimeTracker {
+  private _start: number;
+  private _stopped: boolean = false;
+
+  constructor(private _onFinish: (time: number) => void) {
+    this._start = performance.now();
```

Comment:
AFAIK, `Date.now()` is more performant. And we don't need the precision advantage that `performance.now()` provides.